### ISSUE TITLE
Propagate errors from pwrite/pread correctly

### ISSUE THREAD

**Issue by metan-ucw** [relan/exfat#78]
Writing to a file when filesystem is full returns `EIO` instead of the usual `ENOSPC` this is because we return `EIO` unconditionally in the `libexfat/io.c` in `exfat_generic_write()`. I guess that we should propagate the `errno` from `pread()` and `pwrite()`, following patch fixes the issue for me:

```
diff --git a/libexfat/io.c b/libexfat/io.c
index 63ccfdb..ab18421 100644
--- a/libexfat/io.c
+++ b/libexfat/io.c
@@ -332,7 +332,7 @@ ssize_t exfat_generic_pread(const struct exfat* ef, struct exfat_node* node,
                                        exfat_c2o(ef, cluster) + loffset) < 0)
                {
                        exfat_error("failed to read cluster %#x", cluster);
-                       return -EIO;
+                       return -errno;
                }
                bufp += lsize;
                loffset = 0;
@@ -388,7 +388,7 @@ ssize_t exfat_generic_pwrite(struct exfat* ef, struct exfat_node* node,
                                exfat_c2o(ef, cluster) + loffset) < 0)
                {
                        exfat_error("failed to write cluster %#x", cluster);
-                       return -EIO;
+                       return -errno;
                }
                bufp += lsize;
                loffset = 0;
```

**Comment by metan-ucw** [comment]
I've been to fast to respond here. Looks like the git version is OK while the old version installed on my system is broken. Version 1.2.4 has hardcoded return value for `exfat_truncate()` in `exfat_generic_pwrite()` which produced the `EIO` instead of `ENOSPC`.

Sorry for the noise.

**Comment by relan** [comment]
You are absolutely right, this was a bug. It has been fixed in [v1.2.6](https://github.com/relan/exfat/releases/tag/v1.2.6). Thanks for confirming that `master` works as expected.
